crates/volta-core/src/tool/yarn/resolve.rs

```diff
@@ -72,11 +72,12 @@ fn resolve_semver(matching: VersionReq, hooks: Option<&ToolHooks<Yarn>>) -> Fall
 fn fetch_yarn_index() -> Fallible<(String, PackageIndex)> {
     let url = public_registry_index("yarn");
     let spinner = progress_spinner(&format!("Fetching public registry: {}", url));
-    let metadata: RawPackageMetadata = attohttpc::get(&url)
+    let http_client = Client::new();
+    let metadata: RawPackageMetadata = http_client
+        .get(&url)
         .header(ACCEPT, NPM_ABBREVIATED_ACCEPT_HEADER)
         .send()
-        .and_then(Response::error_for_status)
-        .and_then(Response::json)
+        .and_then(|resp| resp.json())
```

Comment:
Same here, we should maintain the `Response::error_for_status` and the calling style where possible.